osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -190,6 +214,28 @@ protected override void UpdateAfterChildren()
             Ticks.Rotation = Disc.Rotation;
             spmCounter.SetRotation(Disc.RotationAbsolute);
 
+            var newSpins = (int)(Disc.RotationAbsolute / 360) - currentSpins;
+
+            for (int i = currentSpins; i < currentSpins + newSpins; i++)
+            {
+                if (i < 0 || i >= ticks.Count)
+                    break;
+
+                var tick = ticks[i];
+
+                tick.HasBonusPoints = Progress >= 1 && i > Spinner.SpinsRequired;
+
+                if (tick.HasBonusPoints)
+                    bonusCounter
+                        .TransformTextTo($"{(i - Spinner.SpinsRequired) * 1000}")
```

Comment:
As mentioned in the framework PR, we don't want these functions existing (and will probably be removed). You should just set `Text` instead.

Reply:
You could try using `OnRevertResult`, but all this code needs to be more resilient to rewinding in general and you may not need to use it in the end.

Reply:
The way I was thinking this to work in terms of rewinding is that the rotation will decrement and the `currentSpin` would decrement with it, but the problem is that the rotation still increments because we're using `RotationAbsolute`, I proposed that spinners should stick to 1 direction, so you can spin in either direction, but you would stick to it and not change the direction (which no one does that), so that it would be implemented like this:
```csharp
if (direction == 0)
    int direction = Math.Sign(Disc.Rotation * 2); // current rotation is divided by 2 ( (see SpinnerDisc:L128)

int newSpins = (Disc.Rotation * direction / 360) - currentSpins;
```

but that doesn't seem like a good idea since not sure if we would want users to see their progress decrementing when spinning in another direction.

Reply:
You can check whether elapsed time is decreasing or not as a potential avenue for handling this?